src/network/error.rs

```diff
@@ -0,0 +1,15 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct NetavarkError {
+    pub error: String,
```

Comment:
I think we would another field for `errno`. It could be empty or standard `-1` if not specified but we could fill it with number wherever possible.